### ISSUE TITLE
Allow for unauthenticated requests

### ISSUE THREAD

**Issue by thejcannon** [actions/github-script#396]
**Is your feature request related to a problem? Please describe.**
AFAICT GitHub allows unauthenticated requests to public URLs (e.g. `curl -L   https://api.github.com/repos/actions/github-script/pulls` works). 

But currently, passing an empty token results in `| Error: Input required and not supplied: github-token`. 
Passing a garbage token results in: `| RequestError [HttpError]: Bad credentials` from the API

My use-case is trying to test my workflows using https://github.com/nektos/act in a script. Ideally if all of my API calls are public-accessible, we don't need to force callers to provide a PAT.

**Describe the solution you'd like**
Allow me to trudge forward, an anonymous user, scraping the sea of information on the Great GitHub Ocean.

This likely would benefit from being opt-in behind a new `input`. I think the current behavior of erroring if `github-token` is empty is great for catching developer mistakes.

**Describe alternatives you've considered**
Not using this action :cry: 

**Additional context**
<3 this action. Truly powerful, yet elegant.

**Comment by joshmgross** [comment]
👋 @thejcannon,

Thanks for the feature request!

Generally speaking, I don't think we should encourage unauthenticated requests. They're subjected to a lower [rate-limit](https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#rate-limits-for-requests-from-personal-accounts) based on the IP address that you'd likely hit fairly quickly.





> My use-case is trying to test my workflows using https://github.com/nektos/act in a script. Ideally if all of my API calls are public-accessible, we don't need to force callers to provide a PAT.

Could you clarify why providing a PAT isn't necessary? Is it common to use `nektos/act` without passing in some replacement for the Actions `GITHUB_TOKEN`?

Within an Actions workflow, there would always be a token to use. While it may be some friction, I'm reluctant to add an input to this action that's only relevant *outside of* GitHub Actions.

**Comment by thejcannon** [comment]
To be completely honest, I was bracing for this response :joy: 

I can't speak universally for the usage of `act`. In our case we're using it to try and smoke test our GitHub Actions, since we've found that although we can try and shift the actions to be as slim as possible and test the code it calls, the interactions between jobs/steps is still brittle and hard to capture at dev time (e.g. you forgot to use `fromJSON(...)`)

After making this issue (and the subsequent PR) I ran into the aforementioned rate-limit and realized that stubbing out the calls to the public API was all-around a better/safer solution (albeit still leaves some room for error in "prod" code). So, now I type this I realize I can use vanilla GitHub Actions, and pass in a bogus token to satisfy the empty check.

So, I'll leave it up to you/y'all to decide how to handle this issue/PR. I'm completely understanding of these getting closed as "not planned".

Cheers! :tada:
